Update the way run scripts populate the argument parser

cmcantalupo opened the pull request:
There is a lot of repreated code in the run scripts for adding command line arguments.   This change puts this code in a common place in common_args.py.  Another change is to pass the args object directly to the launch functions rather than breaking out the members.  Finally, a basic main() function has been added for experiments to use which takes an app config as input.

dianarg commented on integration/experiment/common_args.py:
```diff
@@ -34,15 +34,36 @@
+
+
+def add_node_count(parser):
+    parser.add_argument('-n', '--node-count', dest='node_count',
```
Do we want to leave open the possibility of forwarding -n to the launch command?  Same question for other short form args.

cmcantalupo replied:
yeah, think we just want to get rid of the short form arguments.

dianarg commented on integration/experiment/power_sweep/power_sweep.py:
```diff
@@ -41,6 +41,21 @@
+from experiment import common_args
+
+def setup_run_args(parser, cool_off_time=60):
+    common_args.add_output_dir(parser)
```
is this deliberately not using common_args.setup_run_args()?

dianarg replied:
if different defaults is the problem, it could forward them to that function (maybe a dict?)

cmcantalupo replied:
promoting the cool_off_time default to be locally configured is the issue that cause the change in the pattern.  I'm not sure its is a huge problem as it is.

cmcantalupo committed:
Add support for command line arguments that apply for all run scripts

- New common_args function setup_run_args() implements common arguments.
- Adds ability to set default values when using add_*() function directly.
- Modifies some existing command line options:
  + --iterations -> --trial-count
  + --nodes -> --node-count
- Adds new common command line options:
  + --step-power
  + --step-frequency
  + --cool-off-time
  + --agent-list

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Update monitor run scripts to use common main()

- Modify launch() caommand to take the args structure directly.
- Call through to common_args.setup_run_args()

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Update power sweep run scripts to use common main()

- Modify launch() command to take the args structure directly.
- Call through to common_args.setup_run_args()

cmcantalupo committed:
Update frequency sweep run scripts to use common main()

- Modify launch() command to take the args structure directly.
- Call through to common_args.setup_run_args()

cmcantalupo committed:
Update energy efficiency run scripts to use common main()

- Modify launch() command to take the args structure directly.

cmcantalupo commented on integration/experiment/common_args.py:
```diff
@@ -34,61 +34,94 @@
+    add_cool_off_time(parser)
+
+
+def add_output_dir(parser, default='.'):
```
Enabling override of defaults upon call to add_*() may not be all that useful since we can always call parser.set_defaults() after the fact.  Maybe we should not make that change.

dianarg replied:
agreed, there should be one way to override defaults

dianarg commented on integration/experiment/energy_efficiency/barrier_frequency_sweep.py:
```diff
@@ -76,28 +82,44 @@ def launch_configs(app_conf_ref, app_conf, default_freq, sweep_freqs, barrier_ha
+                                                        add_turbo_step=True)
+    barrier_hash = geopmpy.hash.crc32_str('MPI_Barrier')
+    default_freq = max(freq_range)
+    targets = launch_configs(app_conf_ref=app_conf_init(add_barriers=False),
```
I'd prefer these still come in as separate AppConf objects, unless we are requiring that all AppConf constructors use the same add_barriers name.  I could see a situation where the app owner chose to have different classes

cmcantalupo replied:
Right, just have two input arguments to launch().  That was a brain fart late yesterday.

cmcantalupo committed:
Address review feedback and failed smoke tests

- Remove default override feature of add_* functions
- Pass to app configuration objects to barrier_frequency_sweep.main
- Fix incorrect import in minife and nekbone frequency sweep
- Fix args usage in run_frequency_sweep_minife.py

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>
